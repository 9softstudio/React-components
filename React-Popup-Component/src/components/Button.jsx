import React, { Component } from 'react'
import PropTypes from 'prop-types';

export default class Button extends Component {
    render() {
        const { text, className, onButtonClick } = this.props;
        return <button className={className} onClick={onButtonClick}>{text}</button>
    }
}

Button.propTypes = {
    text: PropTypes.string,
    className: PropTypes.string,
    onButtonClick: PropTypes.func.isRequired
}
