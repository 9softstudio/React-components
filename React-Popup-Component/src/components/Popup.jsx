import React, { Component } from 'react'
import PropTypes from 'prop-types';
import { mergeClassName } from '../utils'

export default class Popup extends Component {
    renderFooter() {
        const buttons = this.props.buttons.map((button, index) => React.cloneElement(button, { key: index }));
        return <div className="laPopup__footer">{buttons}</div>;
    }

    render() {
        const { popupInnerClassName, title, buttons } = this.props;
        return (
            <div className={mergeClassName(this.props, 'laPopup')}>
                <div className={mergeClassName({}, popupInnerClassName, 'laPopup__inner')} >
                    <div className="laPopup__title">{title}</div>
                    <span className="laPopup__close" toggle="close" onClick={this.props.onCloseClick}>Ã—</span>
                    <div className="laPopup__content">{this.props.children}</div>
                    {buttons && buttons.length > 0 && this.renderFooter()}
                </div>
            </div>
        );
    }
}

Popup.propTypes = {
    onCloseClick: PropTypes.func.isRequired,
    title: PropTypes.string.isRequired,
    buttons: PropTypes.array,
    popupInnerClassName: PropTypes.string
}