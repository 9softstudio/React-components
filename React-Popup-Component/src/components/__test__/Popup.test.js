import React from 'react';
import Popup from '../Popup';
import Button from '../Button';

import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

configure({ adapter: new Adapter() });

describe('<Popup />', () => {
    let title, content, onCloseClick;
    beforeEach(() => {
        title = "this is title";
        content = "this is content";
        onCloseClick = jest.fn();
    });

    test('renders correctly', () => {
        const component = renderer
                    .create(<Popup title={title} onCloseClick={onCloseClick}>{content}</Popup>);
        
        expect(component.toJSON()).toMatchSnapshot();
    });

    test('custom class name', () => {
        const wrapper = shallow(<Popup className="test" title={title} onCloseClick={onCloseClick}>{content}</Popup>);
        expect(wrapper.find('div').first().props().className).toBe('test laPopup');
    });

    test('custom popup inner class name', () => {
        const wrapper = shallow(<Popup popupInnerClassName="test" title={title} onCloseClick={onCloseClick}>{content}</Popup>);
        expect(wrapper.find('div .laPopup__inner').props().className).toBe('test laPopup__inner');
    });

    test('has provide button array: should render footer with number of buttons accordingly', () => {
        const buttons = [
            <Button text="Cancel" className="fake-class-name" onButtonClick={jest.fn()} />,
            <Button text="Create" className="fake-class-name" onButtonClick={jest.fn()} />
        ];
        const wrapper = shallow(<Popup buttons={buttons} title={title} onCloseClick={onCloseClick}>{content}</Popup>);

        expect(wrapper.find('div').last().props().className).toBe('laPopup__footer');
        expect(wrapper.find(Button)).toHaveLength(2);
    })
});