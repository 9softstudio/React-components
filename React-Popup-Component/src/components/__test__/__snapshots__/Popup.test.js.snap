// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Popup /> renders correctly 1`] = `
<div
  className="laPopup"
>
  <div
    className=" laPopup__inner"
  >
    <div
      className="laPopup__title"
    >
      this is title
    </div>
    <span
      className="laPopup__close"
      onClick={[MockFunction]}
      toggle="close"
    >
      Ã—
    </span>
    <div
      className="laPopup__content"
    >
      this is content
    </div>
  </div>
</div>
`;
